docs/reference/api/drivers/SAI.md

```diff
@@ -0,0 +1,52 @@
+
+The bit clock fires on each bit of data transmitted across the data lines, and its frequency is determined by the sampling rate, number of bits per channel and number of channels. The sampling rate is highly device-dependent and needs to be configured for each target (typically found in the target's device.h file).
+
+The word clock runs at the same frequency as the sample rate. It is used to select which active audio channel is being transmitted.
+
+The SAI constructor requires the above pins as well as direction of data transfer (transmitter or receiver). Optionally the SAITransmitter and SAIReceiver classes can be used requiring only the pins.
+
+Note that for full duplex, two objects will need to be instantiated.
+
+### Frame Formats
+| Data Type | Field | Description |
+|-----------|-------|-------------|
+| bool      | bclk_polarity | true for active high                                                     |
+| bool      | wclk_polarity | true for active high                                                     |
+| bool      | ws_delay      | true to toggle ws one bit earlier than the frame                         |
+| uint8_t   | ws_length     | ws assertion length from 1bclk to word length                            |
+| uint8_t   | frame_length  | frame length in word count [1: 32]                                       |
+| uint32_t  | word_mask     | mask on frame for used word (bitfield)                                   |
+| uint8_t   | word_length   | word length in bits [1: 32]                                              |
+| uint8_t   | data_length   | data length within the word. This must be less than or equal to word_length |
+| bool      | lsb_first     | true to send LSB first                                                   |
+| bool      | aligned_left  | true to align left                                                       |
+| uint8_t   | bit_shift     | sample bit shift distance from its alignment point                       |
+
+### Errors
+
+ * `sai_init()` returns `SAI_RESULT_INVALID_PARAM`  if at least one of the given parameters is undefined (NULL)
+ * `sai_init()` returns `SAI_RESULT_ALREADY_INITIALIZED` if SAI is already initialized
+ * `sai_init()` returns `SAI_RESULT_CONFIG_UNSUPPORTED` if the device can never support this configuration
+ * `sai_init()` returns `SAI_RESULT_CONFIG_MISMATCH` if the device is not able to support this configuration at this point in time because of other 'live' constraints
+ (such as a shared format/clock configuration with a sibling)
+ * `sai_free()` does nothing if passed a NULL pointer
+ * `sai_free()` de-initialized & un-clock unused part of the device a device/block can be reinitialized via `sai_init()` after being `sai_free()`d.
```

Comment:
This last description is confusing.  Do you mean that `sai_free()` [is] de-initialized?  If so, say: "[is] de-initialized & the un-clock [is] unused," but this seems like it should be followed by an explanation "if...." like in the previous description ("does nothing [if] passed a NULL pointer).   I can't figure out what 'part of the device' is doing?  Is the un-clock an unused part of the device?  When is it de-initialized & the 'un-clock unused' (this also sounds like a double negative - do you mean 'un-clock'?  and what is that?) . Need clarification, and also needs divisions.  "A device/block can be reinitialized via..." should start a new sentence.

Reply:
This should actually read
```
`sai_free()` de-initializes & un-clocks unused part of the device. A device/block can be reinitialized via `sai_init()` after being `sai_free()`d.
```

Reply:
Ok, I see. This makes more sense now.